src/privacy_analysis/packet_analysis/packet_privacy_port.py

```diff
@@ -1,44 +1,75 @@
 from scapy.layers.inet import TCP, UDP
 from scapy.all import Packet
 from src.dashboard.alerts.alert import Alert, ALERT_TYPE, SEVERITY
 
+# This list is less comprehensive than the nmap scanning list due to the variability of IoT device protocol
+# implementations. The nmap scanning list will alert on a more strict list of open ports, and here will alert on the
+# most suspicious ports that may be open for any device.
 suspicious_ports = [21, 22, 23, 2323, 25, 110, 111, 135]
 
+# A list of suspicious strings to search for in plaintext
+suspicious_strings = ["password", "passwd", "email", "e-mail", "username", "usrname", "api_token"]
 
-# TODO: separate sus ports for TCP/UDP?
 
 class PacketPrivacyPort(PacketPrivacy):
-    def __call__(self, packet: Packet):
-        is_TCP = packet.haslayer(TCP)
-        is_UDP = packet.haslayer(UDP)
+    def __call__(self, packet_input: Packet):
+        self.packet = packet_input
+        is_TCP = self.packet.haslayer(TCP)
+        is_UDP = self.packet.haslayer(UDP)
 
         # Perform TCP and UDP checks
         if is_TCP or is_UDP:
             proto_type = TCP if is_TCP else UDP
 
             # Scan for using port 80 and the plaintext for privacy leaks
-            if (packet[proto_type].dport == 80) or (packet[proto_type].sport == 80):
+            if (self.packet[proto_type].dport == 80) or (self.packet[proto_type].sport == 80):
                 alert_port_80 = Alert(None, "Sending data over unencrypted port.", ALERT_TYPE.PRIVACY, SEVERITY.ALERT)
                 alert_port_80.alert()
-                self.__scan_plaintext(packet, proto_type)
+                self.__scan_plaintext(proto_type)
 
             # Monitor suspicious ports
-            print("Monitoring suspicious ports")
-            if packet[proto_type].dport in suspicious_ports:
-                alert_suspicious_ports = Alert(None, "Suspicious destination port used: " + packet[proto_type].dport,
-                                               ALERT_TYPE.PRIVACY, SEVERITY.WARN)
+            if self.packet[proto_type].dport in suspicious_ports:
+                alert_suspicious_ports = Alert(None, "Suspicious destination port used: " +
+                                               str(self.packet[proto_type].dport), ALERT_TYPE.PRIVACY, SEVERITY.WARN)
                 alert_suspicious_ports.alert()
-                print("Alert on bad port")
 
     # Scan the plaintext for privacy leaks
-    # TODO: regex for email, SSN, credit card
-    def __scan_plaintext(self, packet, proto_type):
+    def __scan_plaintext(self, proto_type):
+        # Try to get the payload
         try:
-            if (packet[proto_type].dport == 80) or (packet[proto_type].sport == 80):
-                print(packet[proto_type].payload)
+            self.payload = str(self.packet[proto_type].payload)
         except:
-            print("plaintext not found")
-            print(packet[proto_type].payload)
             return
```

Comment:
This error handling doesn't really make sense, it should at least log to the error file